### PR TITLE
Course edit page and Institution/Course admin widget.

### PR DESCRIPTION
**Pull request by waheedahmed**
[ECOM-6055](https://openedx.atlassian.net/browse/ECOM-6055)
As a Publisher user, I'd like to edit an existing Parent Course so that I can make changes.

[ECOM-6488](https://openedx.atlassian.net/browse/ECOM-6488)
As a Publisher user, when editing a course, I'd like to associate a Course to an Organization and a Course Admin within that Organization, so that it's clear who is responsible for the Course. 

screencapture-localhost-8004-publisher-courses-15-edit-1485355416701

### CODE REVIEW

**Comment by waheedahmed** [comment]
@bderusha Please review this PR.

**Comment by waheedahmed** [comment]
@tasawernawaz @awais786 Please review.

File: course_discovery/apps/publisher/forms.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 from dal import autocomplete
 from django import forms
 from django.template.loader import render_to_string
+from django.forms import ClearableFileInput
```
Comment (tasawernawaz) [inline course_discovery/apps/publisher/forms.py]: You can use `forms.ClearableFileInput` as we are already importing on line#5

File: course_discovery/apps/publisher/forms.py [hunk 2]
```diff
@@ -29,6 +30,25 @@ def label_from_instance(self, obj):
...
 
 
+class ClearableImageInput(ClearableFileInput):
+    """
+    ClearableFileInput render the saved image as link.
+
+    Render img tag instead of link and add some classes for JS and CSS, also
+    remove image link and clear checkbox.
+    """
+    clear_checkbox_label = '<div class="clear-image"><a class="remove-image" href="#">{remove_text}</a></div>'.format(
```
Comment (tasawernawaz) [inline course_discovery/apps/publisher/forms.py]: Can we put HTML into into own small template and pass context to it.

File: course_discovery/apps/publisher/forms.py [hunk 3]
```diff
@@ -29,6 +30,25 @@ def label_from_instance(self, obj):
...
+        remove_text=_('Remove Image')
+    )
+
+    template_with_initial = (
+        '<img class="course-image-thumbnail" src="%(initial_url)s">'
```
Comment (tasawernawaz) [inline course_discovery/apps/publisher/forms.py]: same here.

**Review by tasawernawaz — APPROVED** [review]
Minor comments

File: course_discovery/apps/publisher/tests/test_views.py [hunk 4]
```diff
@@ -1679,6 +1605,109 @@ def test_edit_page_with_admin(self):
...
+            expected_url=reverse('publisher:publisher_course_detail', kwargs={'pk': self.course.id}),
+            status_code=302,
+            target_status_code=200
+        )
+
+        self.assertEqual(self.course.organizations.first(), organization_extension.organization)
+
+    def _assign_permissions(self, organization_extension):
+        """
+        Dry method to assign permission.
```
Comment (bderusha) [inline course_discovery/apps/publisher/tests/test_views.py]: Don't put `Dry method` in docstrings.  Just say what the method does

File: course_discovery/apps/publisher/tests/test_views.py [hunk 5]
```diff
@@ -1679,6 +1605,109 @@ def test_edit_page_with_admin(self):
...
+        """
+        assign_perm(OrganizationExtension.VIEW_COURSE, organization_extension.group, organization_extension)
+        assign_perm(OrganizationExtension.EDIT_COURSE, organization_extension.group, organization_extension)
+
+    def _post_data(self, organization_extension):
+        """
+        Dry method to generate post data.
```
Comment (bderusha) [inline course_discovery/apps/publisher/tests/test_views.py]: Don't put `Dry method` in docstrings.  Just say what the method does

File: course_discovery/static/js/publisher/publisher.js [hunk 6]
```diff
@@ -57,6 +57,24 @@ $(document).ready(function(){
...
+        $('.course-image-thumbnail').hide();
+        $('.course-image-field a').hide();
+        $('input#image-clear_id').prop('checked', true);
+    });
+
+    // If file selected mark checkbox unchecked otherwise checked.
+    $('input#id_image').change(function (e) {
+        e.preventDefault();
+        if (this.files && this.files[0]) {
+            $('input#image-clear_id').prop('checked', false);
```
Comment (bderusha) [inline course_discovery/static/js/publisher/publisher.js]: Pull this selector into a variable incase in changes since it is used more than once

**Review by bderusha — APPROVED** [review]
A few minor comments 👍 once resolved

**Pull request merged**